From eeaf016b820182d911ea4a4e7cb6651928300599 Mon Sep 17 00:00:00 2001
From: Alberto Milone <alberto.milone@canonical.com>
Date: Tue, 7 Jan 2014 15:30:30 +0100
Subject: [PATCH 1/2] nv-acpi.c: add support for Linux 3.13

There's no need to call acpi_os_wait_events_complete() (or its
NVIDIA equivalent) as, in 3.13, acpi_remove_notify_handler() waits
for events to finish.
---
 nv-acpi.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/nv-acpi.c b/nv-acpi.c
index b0138cd..022950b 100644
--- a/nv-acpi.c
+++ b/nv-acpi.c
@@ -303,7 +303,9 @@ static int nv_acpi_remove(struct acpi_device *device, int type)
 
     if (pNvAcpiObject->notify_handler_installed)
     {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 13, 0)
         NV_ACPI_OS_WAIT_EVENTS_COMPLETE();
+#endif
 
         // remove event notifier
         status = acpi_remove_notify_handler(device->handle, ACPI_DEVICE_NOTIFY, nv_acpi_event);
-- 
1.7.9.5

